tests/ignite/handlers/test_checkpoint.py

```diff
@@ -200,163 +204,162 @@ def test_checkpoint_with_dp():
+    save_handler = MagicMock(spec=BaseSaveHandler)
 
-        trainer = Engine(lambda e, b: None)
-        trainer.state = State(epoch=1, iteration=1, score=0.77)
+    checkpointer = Checkpoint(
+        to_save,
+        save_handler=save_handler,
+        filename_prefix=filename_prefix,
+        global_step_transform=lambda e, _: e.state.epoch,
+    )
 
-        checkpointer(trainer)
-        assert save_handler.call_count == 1
+    trainer = Engine(lambda e, b: None)
+    trainer.state = State(epoch=2, iteration=1)
 
-        metadata = {"basename": name, "score_name": None, "priority": 0.77}
-        save_handler.assert_called_with(obj, f"{name}_0.7700.pt", metadata)
+    checkpointer(trainer)
+    assert save_handler.call_count == 1
 
-        trainer.state.epoch = 12
-        trainer.state.iteration = 1234
-        trainer.state.score = 0.78
+    if len(filename_prefix) > 0:
+        filename_prefix += "_"
 
-        checkpointer(trainer)
-        assert save_handler.call_count == 2
-        metadata["priority"] = 0.78
-        save_handler.assert_called_with(obj, f"{name}_0.7800.pt", metadata)
-        assert save_handler.remove.call_count == 1
-        save_handler.remove.assert_called_with(f"{name}_0.7700.pt")
-        assert checkpointer.last_checkpoint == f"{name}_0.7800.pt"
+    metadata = {"basename": f"{filename_prefix}{name}", "score_name": None, "priority": 2}
+    save_handler.assert_called_with(obj, f"{filename_prefix}{name}_2.pt", metadata)
 
-    model = DummyModel()
-    to_save = {"model": model}
-    _test(to_save, model.state_dict(), "model")
+    trainer.state.epoch = 12
+    trainer.state.iteration = 1234
+    checkpointer(trainer)
+    assert save_handler.call_count == 2
+    metadata["priority"] = 12
+    save_handler.assert_called_with(obj, f"{filename_prefix}{name}_12.pt", metadata)
+    assert save_handler.remove.call_count == 1
+    save_handler.remove.assert_called_with(f"{filename_prefix}{name}_2.pt")
+    assert checkpointer.last_checkpoint == f"{filename_prefix}{name}_12.pt"
+
+
+@pytest.mark.parametrize(
+    "to_save, obj, name",
+    [
+        ({"model": model}, model.state_dict(), "model"),
+        (
+            {"model": model, "optimizer": optimizer},
+            {"model": model.state_dict(), "optimizer": optimizer.state_dict()},
+            "checkpoint",
+        ),
+    ],
+)
+def test_checkpoint_with_score_function(to_save, obj, name):
+    save_handler = MagicMock(spec=BaseSaveHandler)
 
-    model = DummyModel()
-    optimizer = torch.optim.SGD(model.parameters(), lr=0.1)
-    to_save = {"model": model, "optimizer": optimizer}
-    _test(to_save, {"model": model.state_dict(), "optimizer": optimizer.state_dict()}, "checkpoint")
+    checkpointer = Checkpoint(to_save, save_handler=save_handler, score_function=lambda e: e.state.score)
 
+    trainer = Engine(lambda e, b: None)
+    trainer.state = State(epoch=1, iteration=1, score=0.77)
 
-def test_checkpoint_with_score_name_only():
```

Comment:
This test has only one configuration. No need to write it using `pytest.mark.parametrize`

Reply:
okk